R/change.R

```diff
@@ -0,0 +1,98 @@
+#' Calculate changes over a given time period
+#'
+#' Calculates either the absolute or percentage change in a given variable over a given period of time.
+#'
+#' @param df A dataframe.
+#' @param col_name Variable name from the dataframe to calculate the change for.
+#' @param val Chosen value of the variable to calculate the change for. NB the first instance of the value in the column is matched.
+#' @param ago The number of time intervals ago, e.g. quarters or years, to calculate the change from. An integer.
+#' @param type Specify either an "absolute" or "percentage" change.
+#'
+#' @return A numeric value.
+#'
+#' @examples
+#' df <- data.frame("period" = c("2020 Q1", "2020 Q2", "2020 Q3", "2020 Q4"),
+#' "receipts" = seq(10, 40, 10)
+#' )
+#' change(df, "receipts", 40, 2) # returns 20
+#' change(df, "receipts", 40, 1, type = "percentage") # returns "33%"
+#'
+#' @export
+
+change <- function(df, col_name, val, ago, type = "absolute"){
+
+  # Checks on input------------------------------------
+
+  # Check if df is of type dataframe
+  if (!is.data.frame(df)){
+
+    stop("Input to df is not a dataframe")
+  }
+
+  # Check col_name is a character
+  if (!is.character(col_name)){
```

Comment:
Worth noting that most of the time, people will not use quotation marks, so they will get an "object not found" error rather than this helpful error